### PR TITLE
CAMEL-10719 add JMX management to ThrottlingExceptionRoutePolicy

### PR DESCRIPTION
**Pull request by CodeSmell**

### CODE REVIEW

File: camel-core/src/main/java/org/apache/camel/api/management/mbean/ManagedThrottlingExceptionRoutePolicyMBean.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.api.management.mbean;
+
+import org.apache.camel.api.management.ManagedAttribute;
+
+public interface ManagedThrottlingExceptionRoutePolicyMBean extends ManagedServiceMBean {
+
+    @ManagedAttribute(description = "how long to wait before moving open circuit to half open")
```
Comment (davsclaus) [inline camel-core/src/main/java/org/apache/camel/api/management/mbean/ManagedThrottlingExceptionRoutePolicyMBean.java]: Can you upper case the first letter of these descriptions

File: camel-core/src/main/java/org/apache/camel/impl/ThrottlingExceptionRoutePolicy.java [hunk 2]
```diff
@@ -261,11 +261,11 @@ private void logState() {
     
     public String dumpState() {
         int num = state.get();
-        String state = stateAsString(num);
+        String routeState = stateAsString(num);
         if (failures.get() > 0) {
-            return String.format("*** State %s, failures %d, last failure %d ms ago", state, failures.get(), System.currentTimeMillis() - lastFailure);
+            return String.format("*** State %s, failures %d, last failure %d ms ago", routeState, failures.get(), System.currentTimeMillis() - lastFailure);
```
Comment (davsclaus) [inline camel-core/src/main/java/org/apache/camel/impl/ThrottlingExceptionRoutePolicy.java]: Can you remove the ***

Comment (CodeSmell) [inline camel-core/src/main/java/org/apache/camel/impl/ThrottlingExceptionRoutePolicy.java]: I'm on it

Comment (CodeSmell) [inline camel-core/src/main/java/org/apache/camel/impl/ThrottlingExceptionRoutePolicy.java]: Will also add a test for JMX stuff

**Comment by CodeSmell** [comment]
Claus
More of a curiosity thing, but why do the classes in the package _org.apache.camel.impl_ have the corresponding test classes/cases in the package _org.apache.camel.processor_?

File: camel-core/src/main/java/org/apache/camel/impl/ThrottlingExceptionRoutePolicy.java [hunk 3]
```diff
@@ -71,11 +71,11 @@
     private ThrottlingExceptionHalfOpenHandler halfOpenHandler;
 
     // stateful information
-    private final AtomicInteger failures = new AtomicInteger();
-    private final AtomicInteger state = new AtomicInteger(STATE_CLOSED);
-    private volatile Timer halfOpenTimer;
-    private volatile long lastFailure;
-    private volatile long openedAt;
+    private Timer halfOpenTimer;
```
Comment (davsclaus) [inline camel-core/src/main/java/org/apache/camel/impl/ThrottlingExceptionRoutePolicy.java]: Please dont change this back as those changes are for threading, you need to use volatile when using primitives

File: camel-core/src/main/java/org/apache/camel/impl/ThrottlingExceptionRoutePolicy.java [hunk 4]
```diff
@@ -154,31 +157,31 @@ private void calculateState(Route route) {
         
         if (state.get() == STATE_CLOSED) {
             if (failureLimitReached) {
-                LOG.debug("Opening circuit...");
+                LOG.debug("opening circuit...");
```
Comment (davsclaus) [inline camel-core/src/main/java/org/apache/camel/impl/ThrottlingExceptionRoutePolicy.java]: Dont change these back, we dont use lower case logging

File: camel-core/src/main/java/org/apache/camel/impl/ThrottlingExceptionRoutePolicy.java (hunk 3 above)
Comment (CodeSmell) [inline camel-core/src/main/java/org/apache/camel/impl/ThrottlingExceptionRoutePolicy.java]: Sorry. Missed that in the merge when I had to make some other changes for the JMX management

**Comment by CodeSmell** [comment]
The mbarlot is also me, but was working on project code for work and forgot to switch git user when switched over.

**Pull request closed without merging**